pkg/workloads/console/controller.go

```diff
@@ -216,23 +222,50 @@ func (r *reconciler) createOrUpdate(expected recutil.ObjWithMeta, kind string, d
 		return err
 	}
 
-	outcome, err := recutil.CreateOrUpdate(r.ctx, r.client, expected, kind, diffFunc)
+	outcome, err := recutil.CreateOrUpdate(r.ctx, r.client, expected, diffFunc)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "CreateOrUpdate failed")
+	}
+
+	// Use the same 'kind: obj-name' format as in the core controllers, when
+	// emitting events.
+	objDesc := fmt.Sprintf("%s: %s", kind, expected.GetName())
+
+	switch outcome {
```

Comment:
There's a question whether all the callsites to CreateOrUpdate will want to do this sort of logging. It might be beneficial for us to move it into the method itself- can wait until we see another use case though.